lesson17/index.js

```diff
@@ -0,0 +1,161 @@
+const ul = document.getElementById("js-pict-list");
+const previousButton = document.getElementById("js-button-previous");
+const nextButton = document.getElementById("js-button-next");
+
+function addLoading() {
+    const img = document.createElement("img");
+    const imgWrapper = document.createElement("div");
+    img.src = "loading-circle.gif";
+    img.id = "js-loading";
+    imgWrapper.classList.add("loading");
+    ul.appendChild(imgWrapper).appendChild(img);
+}
+
+function removeLoading() {
+    document.getElementById("js-loading").remove();
+}
+
+async function fetchData(api) {
+    const response = await fetch(api);
+    const json = await response.json();
+
+    if(!response.ok){
+        throw new Error(`${response.status}:${response.statusText}`);
+    }
+    return json.data;
+}
+
+async function fetchImgData() {
+    return new Promise((resolve) => {
+        setTimeout(() => resolve(fetchData("https://mocki.io/v1/2db56900-1359-4d61-bd4c-d477c01b0122")), 3000);
+    });
+}
+
+async function getImgData() {
+    addLoading();
+    try {
+        return await fetchImgData();
+    } catch(e) {
+        createErrorMessage(e);
+    } finally {
+        removeLoading();
+    }
+}
+
+function createErrorMessage(error) {
+    const p = document.createElement("p");
+    p.classList.add("slideshow__error");
+    p.textContent = error;
+    console.error(error);
+    ul.appendChild(p);
+}
+
+function createImg(data) {
```

Comment:
IMO
If there is a return value, I think it is appropriate to use create**.
In this case, I think render** is better since we are adding what we created.
```suggestion
function renderImg(data) {
```

Reply:
I had never used the word render before,
so this was a learning experience!
Thank you for the review!
[b63a35c](https://github.com/haru-programming/JavaScript-lessons/pull/21/commits/b63a35cc32db807e4b44a3eb3c2ec62442eb272b)